File: programmers/etc/level1/과일_장수.py
def solution(k, m, score):
    answer = 0
    score.sort(reverse=True)
    for i in range(0, len(score), m):
        tmp = score[i:i+m]
        if len(tmp) == m:
            answer += tmp[m-1]*m
    return answer


k, m = 3, 4
score = [1, 2, 3, 1, 2, 3, 1]
print(solution(k, m, score)) # 8

k, m = 4, 3
score = [4, 1, 2, 2, 4, 4, 4, 4, 1, 2, 4, 2]
print(solution(k, m, score)) # 33

File: Java/programmers/etc/level1/과일장수.java
package CodeTest.Java.programmers.etc.level1;

import java.util.*;

public class 과일장수 {
    public int solution(int k, int m, int[] score) {
        int answer = 0;
        Arrays.sort(score);
            
        for(int i=score.length; i>=m; i-=m){
            answer += score[i-m]*m;
        }

        return answer;
    }
}
